addons/stock/models/stock_move.py

```diff
@@ -605,11 +605,42 @@ def write(self, vals):
             self._propagate_product_packaging(vals['product_packaging_id'])
         if 'date_deadline' in vals:
             self._set_date_deadline(vals.get('date_deadline'))
+        if 'product_id' in vals:
+            orderpoint_domain = [('product_id', 'in', self.product_id.ids)]
+            wh_ids = self.location_id.warehouse_id.ids + self.location_dest_id.warehouse_id.ids
+            if wh_ids:
+                orderpoint_domain.append(('warehouse_id', 'in', wh_ids))
+            self.env.add_to_compute(
+                self.env['stock.warehouse.orderpoint']._fields['qty_to_order'],
+                self.env['stock.warehouse.orderpoint'].search(orderpoint_domain)
+            )
         res = super(StockMove, self).write(vals)
         if move_to_recompute_state:
             move_to_recompute_state._recompute_state()
         if receipt_moves_to_reassign:
             receipt_moves_to_reassign._action_assign()
+        if 'product_id' in vals or 'state' in vals or 'date' in vals or 'product_uom_qty' in vals:
+            orderpoint_domain = [('product_id', 'in', self.product_id.ids)]
+            wh_ids = self.location_id.warehouse_id.ids + self.location_dest_id.warehouse_id.ids
+            if wh_ids:
```

Comment:
This `if` makes sense?

Reply:
Yes, this was added for the case where we have for example a dropshipping move (vendor -> customer location) which has no wh_id set on both locations. In this case we need to fall back to the old behaviour of just updating all the orderpoint on the product. There is a test that uses this type of move where an orderpoint on the WH needs to be updated after a delivery of this type

Reply:
I see, but it means that it is not 100 % correct. When you write/create on multiple moves (X1, X2), X1 doesn't have any warehouse (special case of dropshipping) and X2 is linked to one or more warehouse (normal case), the domain won't be correct: it will filtered out depending warehouse of where it shouldn't (X1 won't be consider). 
Do you know what I mean ?